### ISSUE TITLE
name response spoken when turned draggable

### ISSUE THREAD

**Issue by zepumph** [phetsims/friction#213]
This is because of the default focus listener in Voicing.js is automatically (and with no way to turn off), supplying the name on focus. Perhaps we can figure out a way to not do this in our case.

Caused by https://github.com/phetsims/scenery-phet/issues/693

Tagging @jessegreenberg to note that 24 hours later, we found a case where we want to ignore the newly created focus event for draggable :)

**Comment by zepumph** [comment]
@jessegreenberg and I discussed and thought through two ways forward. We didn't love either, because we don't like duplication of overridding the entire focusListener, and we also want to be able to alter what is spoken, not just IF it is spoken. Here are the two thoughts so far.

```diff

Index: js/accessibility/voicing/Voicing.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/accessibility/voicing/Voicing.js b/js/accessibility/voicing/Voicing.js
--- a/js/accessibility/voicing/Voicing.js	(revision 0687a46cb60b759b6de5230c15886f16fd330133)
+++ b/js/accessibility/voicing/Voicing.js	(date 1628804995753)
@@ -77,6 +77,14 @@
        */
       initializeVoicing( options ) {
 
+        options = merge( {
+          focusListener: () => {
+            this.voicingSpeakFullResponse( {
+              contextResponse: null
+            } );
+          }
+        }, options );
+
         assert && assert( this.voicingInitialized === undefined, 'Voicing has already been initialized for this Node' );
 
         // initialize "super" Trait to support highlights on mouse input
@@ -126,9 +134,11 @@
         // response, and interaction hint.
         this.speakContentOnFocusListener = {
           focus: () => {
-            this.voicingSpeakFullResponse( {
-              contextResponse: null
-            } );
+            if ( this._voicingSpeakOnFocusPredicate() ) {
+              this.voicingSpeakFullResponse( {
+                contextResponse: null
+              } );
+            }
           }
         };
         this.addInputListener( this.speakContentOnFocusListener );

```

**Comment by zepumph** [comment]
Over to @jessegreenberg to brainstorm an alternative or two before discussing further.

**Comment by zepumph** [comment]
Probably instead of having  the default option store the focus listener, we would bind a prototype method, that way we could have usages use `Voicing.defaultFocusListener()` instead of duplicating `speakFullResponse`.

Like this untested thang:


<details>

```diff
Index: js/accessibility/voicing/Voicing.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/accessibility/voicing/Voicing.js b/js/accessibility/voicing/Voicing.js
--- a/js/accessibility/voicing/Voicing.js	(revision 4b7084c2f8bada56121a8d9e291b5b11544a224e)
+++ b/js/accessibility/voicing/Voicing.js	(date 1631113705397)
@@ -39,7 +39,8 @@
   'voicingHintResponse',
   'voicingUtteranceQueue',
   'voicingResponsePatternCollection',
-  'voicingIgnoreVoicingManagerProperties'
+  'voicingIgnoreVoicingManagerProperties',
+  'voicingFocusListener'
 ];
 
 const Voicing = {
@@ -121,15 +122,13 @@
         // are assembled into final content for the UtteranceQueue. See ResponsePatternCollection for more details.
         this._voicingResponsePatternCollection = ResponsePatternCollection.DEFAULT_RESPONSE_PATTERNS;
 
+        this._voicingFocusListener = this.defaultFocusListener.bind( this );
+
         // @private {Object} - Input listener that speaks content on focus. This is the only input listener added
         // by Voicing, but it is the one that is consistent for all Voicing nodes. On focus, speak the name, object
         // response, and interaction hint.
         this.speakContentOnFocusListener = {
-          focus: () => {
-            this.voicingSpeakFullResponse( {
-              contextResponse: null
-            } );
-          }
+          focus: this._voicingFocusListener
         };
         this.addInputListener( this.speakContentOnFocusListener );
 
@@ -473,6 +472,43 @@
       },
       get utteranceQueue() { return this.getUtteranceQueue(); },
 
+      /**
+       * Called whenever this Node is focused.
+       * @public
+       *
+       * @param {function(SceneryEvent):} focusListener
+       */
+      setVoicingFocusListener( focusListener ) {
+        this._voicingFocusListener = focusListener;
+
+        // Input listeners are mutable after adding them to a Node.
+        this.speakContentOnFocusListener.focus = this._voicingFocusListener;
+      },
+
+      set voicingFocusListener( utteranceQueue ) { this.setVoicingFocusListener( utteranceQueue ); },
+
+      /**
+       * Gets the utteranceQueue through which voicing associated with this Node will be spoken.
+       * @public
+       *
+       * @returns {UtteranceQueue}
+       */
+      getVoicingFocusListener() {
+        return this._voicingFocusListener;
+      },
+      get voicingFocusListener() { return this.getVoicingFocusListener(); },
+
+
+      /**
+       * The default focus listener attached to this Node during initialization.
+       * @public
+       */
+      defaultFocusListener() {
+        this.voicingSpeakFullResponse( {
+          contextResponse: null
+        } );
+      },
+
       /**
        * Whether or not a Node composes Voicing.
        * @public

**Comment by zepumph** [comment]
@jessegreenberg and I committed a fix for this, by adding the option to voicing and also by using it in GrabDragInteraction. The main hardship was that we didn't bind the default function specifically so that we could use it in an assertion in GrabDragInteraction. It wasn't the most forward thinking thing, but presumably we can adapt it later if we need to (because it's failing).

I will mark this for me to review the next time I come to this issue.

**Comment by zepumph** [comment]
After another look, I'm still happy with this. Closing
